File: server/src/main/java/cn/net/yunlou/bole/entity/Company.java
package cn.net.yunlou.bole.entity;

import cn.net.yunlou.bole.common.BaseEntity;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import java.util.List;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * FileName: Company Description: Created By MR. WANG Created At 2025/11/24 21:10 Modified By
 * Modified At
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("t_company")
public class Company extends BaseEntity {

    private String name;

    private String email;

    private String holder;

    private String location;

    private String website;

    private String github;

    private String wechat;

    private String bio;

    private Integer followers;

    private Integer fans;

    private Integer likes;

    /** 企业内部评论 */
    @TableField(exist = false)
    private List<CompanyComment> companyComments;
}

File: app/src/types/company.ts
// 查询条件
import { CompanyCommentResult } from "@/types/company-comment";

export interface CompanyQuery extends BodyQuery {
  name?: string;
  status?: number;
  createAt?: [string, string]; // 创建时间范围
}

// 结果
export interface CompanyResult {
  id: number;
  createdAt: string;
  updatedAt: string;
  deleted: number;
  name: string;
  email: string;
  holder: string;
  location: string;
  website: string;
  github: string | null;
  wechat: string | null;
  bio: string | null;
  followers: number;
  fans: number;
  likes: number;
  companyComments: CompanyCommentResult[] | null;
  isFollowed?: boolean;
}

// 表单
export interface CompanyForm {
  name: string;
  email?: string;
  holder?: string;
  location: string;
  website?: string;
  github?: string;
  wechat?: string;
  bio?: string;
}
